src/utils/async.ts

```diff
@@ -174,3 +176,40 @@ export const reduceAsyncIterator = async <T, V>(
     acc = value.reduce(f, acc);
   }
 };
+
+export interface IPage<T> {
+  readonly results: ReadonlyArray<T>;
+  readonly hasMoreResults: boolean;
+}
+
+/**
+ * Maps over an async iterator to get a page of desired size
+ *
+ * @param iter The iterator we want to reduce
+ * @param pageSize The desired page size
+ * @returns an IPage result { results: ReadonlyArray<T>; hasMoreResults: boolean; }
+ */
+export const asyncIteratorToPage = async <T>(
+  iter: AsyncIterator<T | Promise<T>>,
+  pageSize: NonNegativeInteger
+): Promise<IPage<T>> => {
+  const acc = Array<T>();
+  // eslint-disable-next-line functional/no-let
+  let hasMoreResults: boolean = true;
+
+  // eslint-disable-next-line no-constant-condition
+  while (true) {
+    const next = await iter.next();
+    if (next.done === true) {
+      hasMoreResults = false;
+      break;
+    }
+    if (acc.length === pageSize) {
+      break;
+    }
+    // eslint-disable-next-line functional/immutable-data
+    acc.push(await next.value);
+  }
+
+  return { hasMoreResults, results: acc };
+};
```

Comment:
@balanza @BurnedMarshal ?

Reply:
Nothing to say on the implementation, I have some concerns on naming, though.

I'd avoid the concept of Page; instead I'd go for `asyncIteratorToLimitedArray`, also because this method has nearly the same signature ans body of `asyncIteratorToArray`.

Which makes me think: can we collapse both methods into one? Maybe passing the `size` argument as optional with a default value of `Infinity` would make it work.

Reply:
But there is a concept of "page" because we do not return an array, but a model that contains an array and a "paging property" called `hasMoreResult`. I suggested this name to make the intention clear when we use this.

Reply:
Here we are de facto returning a "page of items" from a iterator. Caller's readability is a lot improved with this name.

Reply:
@balanza done on https://github.com/pagopa/io-functions-commons/pull/211/commits/10a4887c90321c1331ea1b0d774ea7d0df593aad